compiler/passes/normalize.cpp

```diff
@@ -376,11 +377,54 @@ static void preNormalizeHandleStaticVars() {
           // Do nothing, that's the default behavior.
         } else if (sharingKind->symbol()->name == astr("computePerLocale")) {
           wrapperVar->addFlag(FLAG_LOCALE_PRIVATE);
+          needsCleanup = true;
         } else {
           USR_FATAL(call, "invalid argument to @functionStatic attribute");
         }
       }
 
+      // If we're a per-locale variable, we'll need to clean it up
```

Comment:
This is absolutely insane but I trust that your code is doing the right thing. This comment is 100% necessary to untangle all the steps going on here.